src/Main.hs

```diff
@@ -334,21 +340,49 @@ inferExpr env exp = let intintint op v1 v2 = do w1 <- checkInt env v1
     ELT _   v1 v2 -> boolintint  LE   v1 v2
     ELEQ _  v1 v2 -> boolintint  LEQ  v1 v2
     EGT _   v1 v2 -> boolintint  GE   v1 v2
-    ETrue _ ->  Ok  $ ExpBool $ LitBool True
-    EFalse _ -> Ok $ ExpBool $ LitBool False
-    -- EITE _ v1 v2 v3 -> do w1 <- checkBool env v1
-    --                       w2 <- checkInt env v2
-    --                       w3 <- checkInt env v3
-    --                       Ok $ ITE w1 w2 w3
+    ETrue _ ->  Ok . ExpBool $ LitBool True
+    EFalse _ -> Ok . ExpBool $ LitBool False
+    -- TODO: make ITE polymorphic
```

Comment:
In this PR or later?

Reply:
Later – there are some problems with it I haven't quite figured out yet